From 3d8caad3f459d8b2136c5bc9be7fdec53e5d777d Mon Sep 17 00:00:00 2001
From: Wu Yongwei <wuyongwei@gmail.com>
Date: Tue, 10 Apr 2018 11:29:12 +0800
Subject: [PATCH] Fix a build problem on Clang.

AND_CAPTURE_MEMBER_FUNCTION_POINTERS workaround had a check for GCC,
but did not exclude Clang.  Clang has a fake GCC version of 4.2, thus
caused problems.
---
 Release/src/http/client/http_client_asio.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Release/src/http/client/http_client_asio.cpp b/Release/src/http/client/http_client_asio.cpp
index 4ba3e085..fca4bb5b 100644
--- a/Release/src/http/client/http_client_asio.cpp
+++ b/Release/src/http/client/http_client_asio.cpp
@@ -47,7 +47,7 @@
 #include <unordered_set>
 #include <memory>
 
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__clang__)
 
 #if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 8)
 #define AND_CAPTURE_MEMBER_FUNCTION_POINTERS
